### ISSUE TITLE
Test scenarios with errors

### ISSUE THREAD

**Issue by MrSent** [tipstrade/TipsTrade-HMRC#7]
Reading the HMRC docs on VAT submissions https://developer.service.hmrc.gov.uk/api-documentation/docs/api/service/vat-api/1.0#_submit-vat-return-for-period_post_accordion there are a number of test scenarios. I've updated to 0.1.9 and TAX_PERIOD_NOT_ENDED isn't there as a scenario. That isn't a problem as I can pass the string through to test it. The question I have really is with regards the error scenarios such as INVALID_MONETARY_AMOUNT and INVALID_NUMERIC_VALUE. These are listed as error responses in the above documentation but I can't seem to test for them. Have I misunderstood how these scenarios work?

**Comment by tipstrade** [comment]
That's been fixed, but only after I pushed 0.0.1.9. It'll be in the next release.

I'll be honest, I put the scenario constants in there, when I first wrote the library, but haven't used them as I was happy to generate my own invalid data when testing (and I was lazy).

The logical method would be to set the specified GovTestScenario, and then inspect the ApiException thrown to ensure that it contains the expected error response. This is probably something that should be done in the unit tests, and I should implement the error constants and the expected error responses in the `GovTestScenarioAttriute` to aid this.

As it is, you could do it by check the `ApiError.Errors` property in the thrown exception:
```C#
var hasExpectedError = ex.ApiError.Errors.Any(x => x.Code == "DUPLICATE_SUBMISSION");
```

**Comment by MrSent** [comment]
Thanks for that.
